vllm_spyre/v1/worker/spyre_worker.py

```diff
@@ -89,8 +90,10 @@ def compile_or_warm_up_model(self) -> None:
             logger.info(
                 "Warming up for prompt length %d, decoding %d tokens with "
                 "batch size %d", prompt_len, num_decode_tokens, batch_size)
-            self._warmup_spyre_fixed_size(prompt_len, num_decode_tokens,
-                                          self.restricted_tokens, batch_size)
+            with _maybe_warmup_context():
```

Comment:
maybe the `with _maybe_warmup_context()` can used one level up before the for loop, instead of importing warmup_mode from torch_sendnn for each warmup shape. Maybe not that important since we don't warmup for many shapes usually.

Reply:
I think it needs to be done on each shape here so that the context closes and runs `update_lazyhandle()` after each shape, right?

Reply:
aah ok you are probably right, I don't really understand what is happening with that context to be honest.
btw, I was looking for `update_lazyhandle` in the code to understand, and couldn't find it, even with a general "find" in the directory. When did that disappear?

Reply:
The call to `update_lazyhandle` is now hidden in the `warmup_mode` context provided by `torch_sendnn` such that it gets called when the context exits. Use of `warmup_mode` was part of the change to remove the deprecated `sendnn_decoder` backend in https://github.com/vllm-project/vllm-spyre/pull/186.

So you'd have to look in the `torch_sendnn` code, but the `warmup_mode` context is basically just:
```
class warmup_mode:
    def __init__(self, enabled: bool = True):
        global _warmup_mode
        self._old_mode = _warmup_mode
        _warmup_mode = enabled

    def __enter__(self):
        return None

    def __exit__(self, exc_type, exc_val, exc_tb):
        global _warmup_mode
        update_lazyhandle()
        _warmup_mode = self._old_mode
 ```